src/libopensc/pkcs15-algo.c

```diff
@@ -448,13 +448,18 @@ static struct sc_asn1_pkcs15_algorithm_info algorithm_table[] = {
 #endif
 #ifdef SC_ALGORITHM_EDDSA
 	/* aka Ed25519 */
-	/* RFC 8410, needed to parse/create X509 certs/pubkeys */
-	{ SC_ALGORITHM_EDDSA, {{1, 3, 101, 112, -1}}, NULL, NULL, NULL },
+	/* RFC 8410, needed to parse/create X509 certs/pubkeys  */
+	/* TODO DEE add asn1_decode_ec_params, asn1_encode_ec_params,asn1_free_ec_params */
+	{ SC_ALGORITHM_EDDSA, {{1, 3, 101, 112, -1}}, NULL, NULL, NULL}, /* Ed25119 */
+	{ SC_ALGORITHM_EDDSA, {{1, 3, 6, 1, 4, 11591, 15, 1, -1}}, NULL, NULL, NULL}, /* OpenPGP */
+	{ SC_ALGORITHM_EDDSA, {{1, 3, 101, 113, -1}}, NULL, NULL, NULL}, /* Ed448 */
 #endif
 #ifdef SC_ALGORITHM_XEDDSA
 	/* aka curve25519 */
-	/* RFC 8410, needed to parse/create X509 certs/pubkeys */
-	{ SC_ALGORITHM_XEDDSA, {{1, 3, 101, 110, -1}}, NULL, NULL, NULL },
+	/* RFC 8410, needed to parse/create X509 certs/pubkeys  no ec_parms*/
+	{ SC_ALGORITHM_XEDDSA, {{1, 3, 101, 110, -1}}, NULL, NULL, NULL}, /* X25519 */
+	{ SC_ALGORITHM_XEDDSA, {{1, 3, 6, 1, 4, 3029, 1, 5, 1 -1}}, NULL, NULL, NULL}, /* OpenPGP */
```

Comment:
Same here, one missing `1`:

http://oidref.com/1.3.6.1.4.1.3029.1.5.1

```suggestion
	{ SC_ALGORITHM_XEDDSA, {{1, 3, 6, 1, 4, 1, 3029, 1, 5, 1 -1}}, NULL, NULL, NULL}, /* OpenPGP curvey25519 */
```

Reply:
Fixing typo and comments